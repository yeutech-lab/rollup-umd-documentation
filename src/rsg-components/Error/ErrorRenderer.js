import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

const styles = ({ fontFamily, fontSize, color, space }) => ({
	root: {
		margin: space[2],
		lineHeight: 1.2,
		fontSize: fontSize.small,
	},
	stack: {
		color: color.error,
		whiteSpace: 'pre',
		fontFamily: fontFamily.monospace,
	},
	message: {
		color: color.error,
		fontFamily: fontFamily.base,
	},
});

export function ErrorRenderer({ classes, error, info }) {
	return (
		<div className={classes.root}>
			<pre className={classes.stack}>
				{error.toString()}
				{info.componentStack.toString()}
			</pre>
			<div className={classes.message}>
				<p>
					This may be due to an error in a component you are overriding, or a bug in React
					Styleguidist.
				</p>
				<p>
					If you believe this is a bug,&nbsp;
					<a
						style={{ color: 'inherit' }}
						href="https://github.com/styleguidist/react-styleguidist/issues"
					>
						please submit an issue
					</a>.
				</p>
			</div>
		</div>
	);
}

ErrorRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	error: PropTypes.object.isRequired,
	info: PropTypes.shape({
		componentStack: PropTypes.any.isRequired,
	}).isRequired,
};

export default Styled(styles)(ErrorRenderer);
