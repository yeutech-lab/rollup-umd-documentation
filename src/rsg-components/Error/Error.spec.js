import React from 'react';
import { ErrorRenderer } from './ErrorRenderer';

it('renderer should render error message', () => {
	const error = { toString: () => 'error' };
	const info = { componentStack: { toString: () => 'info' } };
	const actual = shallow(<ErrorRenderer classes={{}} error={error} info={info} />);

	expect(actual).toMatchSnapshot();
});
