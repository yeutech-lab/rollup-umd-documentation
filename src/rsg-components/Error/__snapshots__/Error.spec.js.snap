// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render error message 1`] = `
<div>
  <pre>
    error
    info
  </pre>
  <div>
    <p>
      This may be due to an error in a component you are overriding, or a bug in React Styleguidist.
    </p>
    <p>
      If you believe this is a bug,Â 
      <a
        href="https://github.com/styleguidist/react-styleguidist/issues"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        please submit an issue
      </a>
      .
    </p>
  </div>
</div>
`;
